### PR TITLE
[IMP] sale: add the qty to deliver in the sale reporting view

### PR DESCRIPTION
**Pull request by chevalierv**
Currently, user needs to extract the report and compute the quantity to deliver through a spreadsheet.

Task #2537839

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/report/sale_report.py [hunk 1]
```diff
@@ -67,6 +68,7 @@ def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
             t.uom_id as product_uom,
             CASE WHEN l.product_id IS NOT NULL THEN sum(l.product_uom_qty / u.factor * u2.factor) ELSE 0 END as product_uom_qty,
             CASE WHEN l.product_id IS NOT NULL THEN sum(l.qty_delivered / u.factor * u2.factor) ELSE 0 END as qty_delivered,
+            CASE WHEN l.product_id IS NOT NULL THEN (product_uom_qty - qty_delivered)/ u.factor * u2.factor ELSE 0 END as qty_to_deliver,
```
Comment (Feyensv) [inline addons/sale/report/sale_report.py]:
```sql
SUM((l.product_uom_qty - l.qty_delivered) / u.factor * u2.factor) ...
```
to keep the grouping logic like the other lines (and that way, you don't have to modify the `groupby_` part).

File: addons/sale_stock/models/sale_order.py [hunk 2]
```diff
@@ -270,7 +270,7 @@ class SaleOrderLine(models.Model):
     free_qty_today = fields.Float(compute='_compute_qty_at_date', digits='Product Unit of Measure')
     qty_available_today = fields.Float(compute='_compute_qty_at_date')
     warehouse_id = fields.Many2one(related='order_id.warehouse_id')
-    qty_to_deliver = fields.Float(compute='_compute_qty_to_deliver', digits='Product Unit of Measure')
+    qty_to_deliver = fields.Float(compute='_compute_qty_to_deliver', digits='Product Unit of Measure', store=True)
```
Comment (Feyensv) [inline addons/sale_stock/models/sale_order.py:273]: :red_circle:

* This is a pure UI field (with its secondary field `display_qty_widget`), it's "clearly" not designed to be stored.
* Two fields are computed in `_compute_qty_to_deliver`, you should never have stored AND non stored fields computed in the same compute (inconsistent states, conflicting recomputations aka unknown danger zone)
* If you investigate `_compute_qty_to_deliver` you will see advanced code not covered by depends, which means it's strictly not expected to be stored, but computed at display time.

**Review by Feyensv — CHANGES_REQUESTED** [review]
The first solution (first commit) is the correct one.

Why did you go for the second solution ? The red runbot should have highlighted it was probably not the best idea :red_circle:

File: addons/sale_stock/models/sale_order.py (hunk 2 above)
Comment (Feyensv) [reply addons/sale_stock/models/sale_order.py:273]: Note, I say "clearly" to insist on the distinction between UI fields and stored data fields, but you'll learn to notice those red flags in the future, I don't blame you for not knowing it was bad to store that field ;).

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**